packages/components/shared/src/controllers/anchor.ts

```diff
@@ -111,10 +111,10 @@ export class AnchorController implements ReactiveController {
 
     if (anchorElement && !this.#host.hasAttribute('aria-details')) {
       anchorElement.id ||= `sl-anchor-${nextUniqueId++}`;
+      this.#host.id ||= `sl-popover-${nextUniqueId++}`;
 
-      this.#host.setAttribute('aria-details', anchorElement.id);
+      anchorElement?.setAttribute('aria-details', this.#host.id);
     }
-
     anchorElement?.setAttribute('aria-expanded', expanded.toString());
 
```

Comment:
I've just recalled that there should also be a focus trap inside the popover when it's still open?

Reply:
I don't think they should have by default. They aren't modal either.
Popover should be a "lighter" interaction than a dialog. If you have a lot of stuff the user needs to navigate through; use a dialog, that has focus trapping (hopefully 😉)

https://hidde.blog/dialog-modal-popover-differences/